kraken-build/src/kraken/std/python/tasks/publish_task.py

```diff
@@ -33,31 +33,28 @@ def get_relationships(self) -> Iterable[TaskRelationship]:
 
     def execute(self) -> TaskStatus:
         credentials = self.index_credentials.get()
-        repository_url = self.index_upload_url.get().rstrip("/") + "/"
         command = [
-            *self.twine_cmd.get(),
-            "upload",
-            "--repository-url",
-            repository_url,
-            "--verbose",
+            "uv",
+            "publish",
+            "--publish-url",
+            self.index_upload_url.get(),
             *[str(x.absolute()) for x in self.distributions.get()],
         ]
-        if credentials:
-            command += [
-                "--username",
-                credentials[0],
-                "--password",
-                credentials[1],
-            ]
-        if not self.interactive.get():
-            command.append("--non-interactive")
         if self.skip_existing.get():
-            command.append("--skip-existing")
+            command.extend(["--check-url", self.index_check_url.get() or self.index_index_url.get()])
+
+        env = os.environ.copy()
+        if credentials:
+            # See https://docs.astral.sh/uv/guides/package/#publishing-your-package
+            # NOTE: PyPI does not support publishing with username and password anymore,
+            #       should we log a warning if the username is not __token__?
+            env["UV_PUBLISH_USERNAME"] = credentials[0]
+            env["UV_PUBLISH_PASSWORD"] = credentials[1]
 
-        safe_command = [x.replace(credentials[1], "MASKED") for x in command] if credentials else command
+        safe_command = command
         self.logger.info("$ %s", safe_command)
 
-        returncode = subprocess.call(command, cwd=self.project.directory)
+        returncode = subprocess.call(command, cwd=self.project.directory, env=env)
```

Comment:
nit: would prefer this style, as it keeps the intent of "we need to pass the full environment not just overrides" more localised

```suggestion
        returncode = subprocess.call(command, cwd=self.project.directory, env={**os.environ, **env})
```

Reply:
env is initialized with os.environ so this would have no effect surely?